src/subcommands/lexical_search.go

```diff
@@ -60,15 +61,59 @@ func LexicalSearch(args LexicalSearchArgs) ([]fileOccurrence, error) {
 		}
 	}
 
-	for _, occList := range newSearchMap {
-		sort.Slice(occList, func(i, j int) bool {
-			return occList[i].Occurrence > occList[j].Occurrence
+	newStemmedSearchMap := make(searchMap)
+	for _, entry := range index.Entries {
+		for word, occ := range entry.StemmedWordOccurrences {
+			newFileOcc := fileOccurrence{
+				FileName:   entry.Name,
+				Occurrence: occ,
+			}
+			if val, ok := newStemmedSearchMap[word]; ok {
+				newStemmedSearchMap[word] = append(val, newFileOcc)
+			} else {
+				newOccList := []fileOccurrence{newFileOcc}
+				newStemmedSearchMap[word] = newOccList
+			}
+		}
+	}
+
+	term := args.SearchTerm
+	stemmedTerm, err := snowball.Stem(term, "english", true)
+	if err != nil {
+		return nil, fmt.Errorf("failed to stem word: %w", err)
+	}
+
+	stemmedResults := newStemmedSearchMap[stemmedTerm]
+	nonStemResults := newSearchMap[term]
+
+	// combine stemmed and non-stemmed occurrence numbers
```

Comment:
Make this comment more detailed, explain the "why" not the "what"

Reply:
done